zio/detector/file.go

```diff
@@ -168,3 +169,55 @@ func OpenFiles(zctx *resolver.Context, dir zbuf.RecordCmpFn, paths ...string) (*
 	}
 	return zbuf.NewCombiner(readers, dir), nil
 }
+
+type inOrderFileReader struct {
+	reader *zbuf.File
+	zctx   *resolver.Context
+	paths  []string
+	cfg    OpenConfig
+}
+
+// InOrderFileReader is a zbuf.ReadCloser compliant reader
+// that takes an array paths and in-order reads all records in each
+// file.
```

Comment:
```suggestion
// SequentialFileReader returns a zbuf.ReadCloser that reads the records from the provided slice of files sequentially.
```

(and along with this suggestion, rename InOrderFileReader to SequentialFileReader)

Reply:
sounds good

Reply:
Given the behavioral similarity to io.MultiReader, I think we should call this MultiFileReader.

Reply:
Oooh kind of nice. @henridf what do you think?

Reply:
I made zbuf.MultiReader at one point but got rid of it.

Reply:
Fancy!

Reply:
In this thread it just keeps on getting better!